### ISSUE TITLE
Compare different cost functions

### ISSUE THREAD

**Issue by tyronerees** [fitbenchmarking/fitbenchmarking#491]
Find and implement a metric which allows users to compare the success of different cost functions to fit their data in one table.

**Comment by tyronerees** [comment]
@jfowkes suggested that calculating https://en.wikipedia.org/wiki/Akaike_information_criterion or https://en.wikipedia.org/wiki/Bayesian_information_criterion and presenting these in a results table might help here.

**Comment by AndrewLister-STFC** [comment]
For our problem, I don't think the information criteria will work. These seem to be more relevent when selecting between ML models or when trying to figure out the model rather than just how we define the error.

**Comment by AndrewLister-STFC** [comment]
@tyronerees and I discussed this and decided that some kind of visual comparison would be better.

Perhaps a graph of all minimizers with cost function x in red and all minimizers with cost function y in blue.
I think this needs a bit more thought still to figure out a good way of presenting this to the user.

**Comment by jfowkes** [comment]
> For our problem, I don't think the information criteria will work. These seem to be more relevent when selecting between ML models or when trying to figure out the model rather than just how we define the error.

Agreed. As the different cost functions will typically have the same number of parameters, AIC/BIC will just collapse to the negative log-likelihood which are just the cost functions.

**Comment by AndrewLister-STFC** [comment]
> 
> 
> @tyronerees and I discussed this and decided that some kind of visual comparison would be better.
> 
> Perhaps a graph of all minimizers with cost function x in red and all minimizers with cost function y in blue.
> I think this needs a bit more thought still to figure out a good way of presenting this to the user.

Something like this... but not drawn in paint 
![image](https://user-images.githubusercontent.com/48282025/105853817-04a17280-5fde-11eb-9aba-ea3fbbc8c24c.png)

**Comment by jfowkes** [comment]
After discussions with @tyronerees, one of the following might be a good idea:
https://en.wikipedia.org/wiki/Cross-validation_(statistics) 
e.g. LOOCV or k-fold are fairly common aprroaches.

**Comment by AndrewLister-STFC** [comment]
> 
> 
> After discussions with @tyronerees, one of the following might be a good idea:
> [en.wikipedia.org/wiki/Cross-validation_(statistics)](https://en.wikipedia.org/wiki/Cross-validation_(statistics))
> e.g. LOOCV or k-fold are fairly common approaches.

I've only seen cross-validation used as an option in data sparse environments (ML) in which one wishes to build some kind of predictive model while avoiding overfitting.

In our case I again think that the models are very unlikely to overfit as the models are provided as part of the problem, usually with a very small number of params.

I think something similar could be useful, although we also need to be careful to make sure that we're consistent both between minimizers on a problem, and with a users expectation (i.e. if a user trys to solve the problem outside of fitbenchmarking, they should get the same results).

**Comment by jfowkes** [comment]
I think from a user perspective the visual comparison is definitely the way to go.

It may also be useful to report something like LOOCV/k-fold to get a sense of how robust each model is, but I agree we are extremely unlikely to overfit with our limited parameter set.

**Comment by AndrewLister-STFC** [comment]
After talking to Anthony earlier in the week, he told me that they ran cross validation on simulated data to check cost functions within Mantid.

Specifically they created `n` simulated datasets `y_i` for some fixed input `x`, s.t. the only difference arose from the random noise (poisson), then fit for each to produce a set of parameters `p_i`.

They then calculated the mean and std deviation of the `p_i`.

I think the size of `x` vs. the std dev is what's shown in the graph [here](https://docs.mantidproject.org/nightly/fitting/fitcostfunctions/Poisson.html), although I'm not sure of the size of `n` in these.

@AnthonyLim23 can you confirm that I've got that right?

**Comment by AnthonyLim23** [comment]
Yes, so the x axis is just the number of events in each simulated run. The y axis is the s.d. of the parameter

**Comment by tyronerees** [comment]
There are some statistics mentioned in the Matlab Stats toolbox (https://uk.mathworks.com/help/stats/nonlinearmodel-class.html) which may be of interest here: 

![image](https://user-images.githubusercontent.com/13645545/121376143-2e2dbb80-c939-11eb-95e2-67e6304da745.png)

**Comment by AnthonyLim23** [comment]
looks like the leverage only looks like it only identifies if the x value is an outlier https://online.stat.psu.edu/stat462/node/170/
